src/node.cc

```diff
@@ -4531,6 +4531,8 @@ inline int Start(Isolate* isolate, IsolateData* isolate_data,
     } while (more == true);
   }
 
+  AsyncWrap::RunDestroyCbs(&env);
```

Comment:
If `destroy` creates a new asynchronous resource (think logging) then this won't be executed here. This is why I think we should treat the `destroy` queue exactly like the `setImmediate` queue, by using `uv_check_start`.

I also don't like to pollute the event loop code if it can be handled by libuv.